### PR TITLE
[FSSDK-10665] fix: Github Actions YAML files vulnerable to script injections corrected

### PR DESCRIPTION
**Pull request by FarhanAnjum-opti**
## Summary
Instead of directly using variables that can contain user input, new intermediate variables were introduced so that script injection attacks could be mitigated. Also **default environment variables** were used wherever possible.

## Issues
- [FSSDK-10665](https://jira.sso.episerver.net/browse/FSSDK-10665)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,20 +32,21 @@ jobs:
...
             -H "Accept: application/vnd.github+json" \
             -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
-            "${{ github.api_url }}/repos/${{ github.repository }}/releases/latest" \
+            "$GITHUB_API_URL/repos/$GITHUB_REPOSITORY/releases/latest" \
           | jq -r .tag_name)" >> $GITHUB_OUTPUT
 
       - id: npm-tag
         name: Determine NPM tag
+        env:
+          ALTERNATE_RELEASE_TAG: ${{ github.event.release.tag_name }}
```
Comment (raju-opti) [inline .github/workflows/release.yml:41]: Can we rename it to GITHUB_RELEASE_TAG instead?

**Review by raju-opti — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (FarhanAnjum-opti) [reply .github/workflows/release.yml:41]: Sure. Renaming it.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -32,20 +32,21 @@ jobs:
           echo "latest-release-tag=$(curl -qsSL \
             -H "Accept: application/vnd.github+json" \
             -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
-            "${{ github.api_url }}/repos/${{ github.repository }}/releases/latest" \
+            "$GITHUB_API_URL/repos/$GITHUB_REPOSITORY/releases/latest" \
```
Comment (mikechu-optimizely) [inline .github/workflows/release.yml:35]: Are GITHUB_API_URL and GITHUB_REPOSITORY some defaults in all workflows or do we have to set them in an `env:` block?

**Review by raju-opti — APPROVED** [review]
LGTM

**Pull request merged**